pkg/cmd/adm/restart_test.go

```diff
@@ -1,216 +1,390 @@
+			func(ctx *clicontext.CommandContext, f cmdutil.Factory, ioStreams genericclioptions.IOStreams, labelSelector string) error {
+				return expectedErr
+			}, nil)
+
+		//then
+		require.EqualError(t, err, expectedErr.Error())
+	})
 
-		// when
-		err := restartHostOperator(ctx, fakeClient, cfg.OperatorNamespace)
+	t.Run("autoscalling deployment should not restart", func(t *testing.T) {
+		//given
+		newClient, fakeClient := NewFakeClients(t, toolchainCluster, memberDeployment, autoscalarDeployment)
+		ctx := clicontext.NewCommandContext(term, newClient)
+		//when
+		err := restartDeployment(ctx, fakeClient, "toolchain-member-operator", nil, genericclioptions.NewTestIOStreamsDiscard(),
+			func(ctx *clicontext.CommandContext, f cmdutil.Factory, ioStreams genericclioptions.IOStreams, labelSelector string) error {
+				return nil
+			}, nil)
 
-		// then
-		require.Error(t, err)
-		AssertDeploymentHasReplicas(t, fakeClient, namespacedName, 1)
-		assert.Equal(t, 0, numberOfUpdateCalls)
+		//then
+		require.NoError(t, err)
+		require.Contains(t, term.Output(), "Found only autoscaling-buffer deployment in namespace toolchain-member-operator , which is not required to be restarted")
 	})
 
-	t.Run("there are more deployments with the host operator label - restart fails", func(t *testing.T) {
-		// given
-		deployment := newDeployment(namespacedName, 1)
-		deployment.Labels = map[string]string{"olm.owner.namespace": "toolchain-host-operator"}
-		deployment2 := deployment.DeepCopy()
-		deployment2.Name = "another"
-		newClient, fakeClient := NewFakeClients(t, deployment, deployment2)
-		numberOfUpdateCalls := 0
-		fakeClient.MockUpdate = requireDeploymentBeingUpdated(t, fakeClient, namespacedName, 1, &numberOfUpdateCalls)
+}
+
+func TestRestart(t *testing.T) {
+
+	t.Run("restart should start with y response", func(t *testing.T) {
+		//given
+		SetFileConfig(t, Host())
+		toolchainCluster := NewToolchainCluster(ToolchainClusterName("host"))
+		deployment := newDeployment(test.NamespacedName("toolchain-host-operator", "host-operator-controller-manager"), 1)
+		deployment.Labels = make(map[string]string)
+		deployment.Labels["kubesaw-control-plane"] = "kubesaw-controller-manager"
+		term := NewFakeTerminalWithResponse("Y")
+		newClient, _ := NewFakeClients(t, toolchainCluster, deployment)
 		ctx := clicontext.NewCommandContext(term, newClient)
 
-		// when
-		err := restartHostOperator(ctx, fakeClient, cfg.OperatorNamespace)
+		//when
+		err := restart(ctx, "host")
 
-		// then
-		require.Error(t, err)
-		AssertDeploymentHasReplicas(t, fakeClient, namespacedName, 1)
-		assert.Equal(t, 0, numberOfUpdateCalls)
+		//then
+		require.ErrorContains(t, err, "no such host") //we expect an error as we have not set up any http client , just checking that it passes the cmd phase and restartdeployment method is called
+		require.Contains(t, term.Output(), "Fetching the current Operator and non-Operator deployments of the operator in")
 	})
+
 }
```

Comment:
I would either drop this test or mock the `restartDeployment` function. Expecting an error for a valid input is not the best practice

Reply:
okay , makes sense